libdnf/utils/utils.cpp

```diff
@@ -24,8 +31,95 @@ std::vector<std::string> string::split(const std::string &source, const char *de
     return container;
 }
 
+std::vector<std::string> string::rsplit(const std::string &source, const char *delimiter, int maxSplit)
+{
+    if (source.empty())
+        throw "Source cannot be empty";
+
+    std::string sequence = source;
+    std::string::size_type tokenBeginIndex = 0;
+    std::vector<std::string> container;
+
+    while ((tokenBeginIndex = sequence.find_last_of(delimiter)) != std::string::npos) {
+        if (maxSplit != -1 && ((int) (container.size() + 1 )) == maxSplit) {
+            container.emplace_back(source.substr(0, tokenBeginIndex));
+            break;
+        }
+
+        container.emplace(container.begin(), source.substr(tokenBeginIndex + 1));
+        sequence = sequence.substr(0, tokenBeginIndex);
+    }
+
+    if (container.empty()) {
+        throw "No delimiter found in source: " + source;
+    }
+
+    return container;
+}
+
+std::string string::trimSuffix(const std::string &source, const std::string &suffix)
+{
+    if (source.length() < suffix.length())
+        throw "Suffix cannot be longer than source";
+
+    if (string::endsWith(source, suffix))
+        return source.substr(0, source.length() - suffix.length());
+
+    throw "Suffix '" + suffix + "' not found";
+}
+
+std::string string::trimPrefix(const std::string &source, const std::string &prefix)
+{
+    if (source.length() < prefix.length())
+        throw "Prefix cannot be longer than source";
+
+    if (string::startsWith(source, prefix))
+        return source.substr(prefix.length() - 1);
+
+    throw "Prefix '" + prefix + "' not found";
+}
+
+bool string::startsWith(const std::string &source, const std::string &toMatch)
+{
+    return source.find(toMatch) == 0;
+}
+
+bool string::endsWith(const std::string &source, const std::string &toMatch)
+{
+    auto it = toMatch.begin();
+    return source.size() >= toMatch.size() &&
+           std::all_of(std::next(source.begin(), source.size() - toMatch.size()), source.end(), [&it](const char &c) {
+               return c == *(it++);
+           });
+}
+
 bool filesystem::exists(const std::string &name)
 {
     struct stat buffer;
     return stat(name.c_str(), &buffer) == 0;
-}
\ No newline at end of file
+}
+
+std::vector<std::string> filesystem::getDirContent(const std::string &dirPath)
+{
+    std::vector<std::string> content;
+    struct dirent *ent;
+    DIR *dir = opendir(dirPath.c_str());
+
+    if (dir != nullptr) {
+        while ((ent = readdir(dir)) != nullptr) {
+            if (strcmp(ent->d_name, "..") == 0 ||
+                    strcmp(ent->d_name, ".") == 0 )
+                continue;
+
+            auto fullPath = dirPath;
+            if (!string::endsWith(fullPath, "/"))
+                fullPath += "/";
+            fullPath += ent->d_name;
+
+            content.emplace_back(fullPath);
+        }
+        closedir (dir);
+    }
+
+    return content;
```

Comment:
this would return garbage if opendir() fails, won't it?

Reply:
It would return empty vector.